gomock/controller_test.go

```diff
@@ -187,11 +239,68 @@ func TestUnexpectedArgCount(t *testing.T) {
 	reporter.assertFatal(func() {
 		// This call is made with the wrong number of arguments...
 		ctrl.Call(subject, "FooMethod", "argument", "extra_argument")
-	})
+	}, "no matching expected call", "Invalid number of arguments of call", "Set: 2, while this call takes: 1")
 	reporter.assertFatal(func() {
 		// ... so is this.
 		ctrl.Call(subject, "FooMethod")
+	}, "no matching expected call", "Invalid number of arguments of call", "Set: 0, while this call takes: 1")
+	reporter.assertFatal(func() {
+		// The expected call wasn't made.
+		ctrl.Finish()
 	})
+}
+
+func TestExpectedMethodCall_CustomStruct(t *testing.T) {
```

Comment:
Added 2 comments. If you find this test not needed, I may as well remove it.
   * comment to a similar, but easier test: https://github.com/golang/mock/pull/97/files#diff-763a69b80e2e0fb14b7d265a0f71d31eR190
  * comment to this test: https://github.com/golang/mock/pull/97/files#diff-763a69b80e2e0fb14b7d265a0f71d31eR249